tsl/src/continuous_aggs/common.c

```diff
@@ -582,21 +582,26 @@ get_bucket_width(CAggTimebucketInfo bucket_info)
 			break;
 		case INTERVALOID:
 		{
+			/*
+			 * Original interval should not be changed, hence create a local copy
+			 * for this check.
+			 */
+			Interval interval = { .time = bucket_info.interval->time,
+								  .day = bucket_info.interval->day,
+								  .month = bucket_info.interval->month };
```

Comment:
```suggestion
			Interval interval = *bucket_info.interval;
```
As per @pmwkaa suggested.